docs/configuration-reference.md

```diff
@@ -1812,6 +1812,25 @@ basic_auth:
 # read from the configured file. It is mutually exclusive with `bearer_token`.
 [ bearer_token_file: /path/to/bearer/token/file ]
 
+# Configures SigV4 request signing. The default credentials chain will be used,
+# documented here:
+#
+# https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
+#
+# When enabled, region must be supplied.
+#
+# This feature is currently exclusive to the Grafana Cloud Agent and is only
+# currently available for remote_write of Prometheus metrics.
+sigv4:
+  # Enable SigV4 request signing. May not be enabled at the same time as
+  # configuring basic auth or bearer_token/bearer_token_file.
+  [ enabled: <boolean> | default = false ]
+
+  # Region to use for signing the requests. When sigv4.enabled is true,
+  # must be non-empty and must be the region of the AMP workspace specified
+  # by the remote_write URL.
+  region: <string>
```

Comment:
Do we have the concept of vendor specific setup? IE should this be
- aws:
   - sig4:
      - region:
      Though not super happy with how that looks above

Reply:
Hmm, that's a good question. This is the first vendor-specific code we've ever added, but given SigV4 implies AWS, I'm not sure if it needs another level of nesting.